### PR TITLE
Add runtime type validation using runtypes

### PR DESCRIPTION
**Pull request by broerjuang**
I add runtypes to documenting what is returned to API. So maybe it helps. 

- [x] adding runtypes
- [x] add guard on `hasQuota` and `getSchedule`

### CODE REVIEW

File: src/data/schedule.ts
Comment (broerjuang) [inline src/data/schedule.ts]: Some quota fields has empty record.

File: src/pages/index.tsx
Comment (broerjuang) [inline src/pages/index.tsx]: change the schedule to locations. Not sure if this one is right choice 😅

File: src/data/location.ts
Comment (broerjuang) [inline src/data/location.ts]: at first I think that I want to throw error here if the contract is different. But since the API will be added or remove very fast. Just log this one into the console.

**Comment by pveyes** [comment]
cc-ing @resir014 as it's related to #26

**Comment by resir014** [comment]
@broerjuang @pveyes Let's wait until #34 is merged, since that's the farthest we have with the types. We can focus this PR solely on runtypes validation.

**Review by resir014 — CHANGES_REQUESTED** [review]
#34 has been merged. Please merge from main and use the types from there as the base. We'll focus this PR solely on adding `runtypes`.

**Comment by broerjuang** [comment]
Damn, I just force push this. thinking that no-one use this branch :( how do i revert it @resir014

**Comment by broerjuang** [comment]
@resir014 been rebased it to master and adding just runtypes to this PR.

File: src/helpers/QuotaHelpers.ts
Comment (pveyes) [inline src/helpers/QuotaHelpers.ts]: does this also validates that each kuota > 0?

Previously the logic uses boolean cast, but the actual intention is to handle == 0

Comment (broerjuang) [reply src/helpers/QuotaHelpers.ts]: oh ya, this one is missing :( will update revert it to the latest one. It just check the existence of the fields.

**Comment by resir014** [comment]
@broerjuang ESLint errors, please fix.

**Comment by pveyes** [comment]
sorry can you wait for #61 & rebase afterwards 🙇

**Comment by broerjuang** [comment]
rebase it already, and change `jakiKuota` to string. Since the value still returned from the server.  🙇

**Review by pveyes — APPROVED** [review]
LGTM just need a green light from @resir014

**Review by resir014 — APPROVED** [review]
All good. Thanks!

**Pull request merged**
